### ISSUE TITLE
Get familiar with Mac's notarization process

### ISSUE THREAD

**Issue by jcagme** [dotnet/arcade#3573]
Understand what the goal of it is and how does it work. Check what tools are needed and what do they do. 

Quick list of things I have to check:

* Does `dotnet` has entitlements?
* How are we signing it?
* What happens if I run `xcrun altool --notarize-app` against it?
* What are other teams doing?

**Comment by jcagme** [comment]
After investigations and talking to folks we've come with the following tactical plan:

**Product teams**

1. Add a plist which sets CSFlags to 65536. This is used to preserve the codesign flag “-o runtime” when codesign is called on it. We’d need to do `ld -sectcreate _TEXT __info_plist  plist_name.plist`
1. Add entitlements. `codesign –entitlements <entitlements>  <asset>`. Here we need two things:
i. Define what entitlements are needed. This enumerates all the supported options. From what I saw I think we need something like:
```xml
    <key>com.apple.security.cs.allow-jit</key>
    <true/>
    <key>com.apple.security.cs.allow-unsigned-executable-memory</key>
    <true/>
    <key>com.apple.security.cs.disable-library-validation</key>
    <true/>
```
ii.  Determine what binaries need to be entitled

**Eng team**

1. Expand the pkgs, zip the native binaries and executables and send the zips to ESRP. With this, binaries will be signed and hardened 
1. Unzip the files, put binaries/executables in their original place and repack
1. Send the pkg for notarization with `xcrun altool --notarize-app --primary-bundle-id "com.dotnet.core.sdk" --username "AppleIdUserName" --password "AppleIdPassword –file dotnet-runtime-3.0.0-preview7-27912-14-osx-x64.pkg` (we could also use a script that the Xamarin team shared here). I don’t think we have an Apple id but I guess is easy to get one. If we call altool today against the SDK we get this and this against the runtime
1. If the pkg is notarized correctly, push the packages to the feed as we do today.

**Comment by jcagme** [comment]
Eng team will own https://github.com/dotnet/arcade/issues/3665

**Comment by markwilkie** [comment]
Is there a tracking issue for the work in vseng?

**Comment by jcagme** [comment]
We've identified the steps that the mechanism to notarize assets has to include. And are described bellow. Closing this continuing to track progress on #3663 
1.	Receive a pkg/app as an input
2.	Unpack ALL the containers in 1
3.	Identify what things are signable and copy them to a folder
4.	Use the ESRP VSO task to get the files in 3 signed (if these files already have entitlements, they’ll also be hardened)
5.	Once ESRP has finished signing, replace the original files in 3 with the signed version
6.	Repack the containers
7.	Send the package to the notarization service
8.	If notarization was approved, publish the package, if not fail the build
